pkg/sync/sync.go

```diff
@@ -67,7 +61,38 @@ func CopyFile(client SyncClient, localPath string, compInfo common.ComponentInfo
 		}
 
 		tarWriter.Close()
+	}()
+
+	return reader, err
+}
+
+// CopyFile copies localPath directory or list of files in copyFiles list to the directory in running Pod.
+// copyFiles is list of changed files captured during `odo watch` as well as binary file path
+// During copying binary components, localPath represent base directory path to binary and copyFiles contains path of binary
+// During copying local source components, localPath represent base directory path whereas copyFiles is empty
+// During `odo watch`, localPath represent base directory path whereas copyFiles contains list of changed Files
+func CopyFile(client SyncClient, localPath string, compInfo common.ComponentInfo, targetPath string, copyFiles []string, globExps []string) error {
+
+	// Destination is set to "ToSlash" as all containers being ran within OpenShift / S2I are all
+	// Linux based and thus: "\opt\app-root\src" would not work correctly.
+	dest := filepath.ToSlash(filepath.Join(targetPath, filepath.Base(localPath)))
+	targetPath = filepath.ToSlash(targetPath)
+
+	klog.V(4).Infof("CopyFile arguments: localPath %s, dest %s, targetPath %s, copyFiles %s, globalExps %s", localPath, dest, targetPath, copyFiles, globExps)
+	reader, writer := io.Pipe()
+	// inspired from https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cp.go#L235
+	go func() {
+		defer writer.Close()
 
+		tarWriter := tar.NewWriter(writer)
+
+		err := makeTar(localPath, dest, copyFiles, globExps, tarWriter)
+		if err != nil {
+			log.Errorf("Error while creating tar: %#v", err)
+			os.Exit(1)
```

Comment:
Did you mean to keep the log & os.Exit here, or should it just be changed to a `return errors.Wrap(err, "error while creating tar")`

Reply:
Same as copy file